util/pyclient/test_client.py

```diff
@@ -138,81 +134,60 @@ def testRetry(self):
 
     async def _testRetry(self):
         """Start servers after a delay in parallel with a write request"""
-        try:
-            async with trio.open_nursery() as nursery:
-                nursery.start_soon(self.startServersWithDelay)
-                nursery.start_soon(self.writeWithRetryAssert)
-        except:
-            raise
-        finally:
-            self.stopServers()
+        await self.writeWithRetryAssert()
 
     async def writeWithRetryAssert(self):
         """Issue a write and ensure that a retry occurs"""
-        config = self.config._replace(req_timeout_milli=5000)
-        val = 1
+        config = self.config._replace(retry_timeout_milli=0)
+        val = 2
         with bft_client.UdpClient(config, self.replicas, None) as udp_client:
            self.assertEqual(udp_client.retries, 0)
-           await udp_client.sendSync(self.writeRequest(val), False)
+           try:
+            await udp_client.sendSync(self.writeRequest(val), False)
+           except(trio.TooSlowError):
+               pass
            self.assertTrue(udp_client.retries > 0)
 
-    async def startServersWithDelay(self):
-        # Retry timeout is 50ms
-        # This guarantees we wait at least one retry with high probability
-        await trio.sleep(.250)
-        self.startServers()
-
+    @verify_system_is_ready
     def testPrimaryWrite(self):
         """Test that we learn the primary and using it succeeds."""
-        self.startServers()
-        try:
-            trio.run(self._testPrimaryWrite)
-        except:
-            raise
-        finally:
-            self.stopServers()
-
-    @verify_system_is_ready_before_starting
+        trio.run(self._testPrimaryWrite)
+
+    @verify_system_is_ready
     async def _testPrimaryWrite(self):
        # Try to guarantee we don't retry accidentally
        config = self.config._replace(retry_timeout_milli=500)
        with bft_client.UdpClient(config, self.replicas, None) as udp_client:
            self.assertEqual(None, udp_client.primary)
-           await udp_client.sendSync(self.writeRequest(1), False)
+           await udp_client.sendSync(self.writeRequest(3), False)
```

Comment:
are you sure the unittest framework will always run the test cases in the same order?

Reply:
It doesn't really matter.
I just wanted to have different values for different tests